flow/transactions/sign/sign.go

```
@@ -54,65 +57,51 @@ var Cmd = &cobra.Command{
-			err   error
+			tx             *flow.Transaction
+			allAuthorizers []string = []string{}
+			payer          flow.Address
+			err            error
 		)
 
-		if conf.Code != "" {
-			code, err = ioutil.ReadFile(conf.Code)
-			if err != nil {
-				cli.Exitf(1, "Failed to read transaction script from %s", conf.Code)
-			}
-		}
-
 		if conf.PayerAddress != "" {
 			payer = flow.HexToAddress(conf.PayerAddress)
 		} else {
 			payer = signerAccount.Address
 		}
 
-		tx := flow.NewTransaction().
-			SetScript(code)
-
-		// Arguments
-		if conf.Args != "" {
-			transactionArguments, err := cli.ParseArguments(conf.Args)
-			if err != nil {
-				cli.Exitf(1, "Invalid arguments passed: %s", conf.Args)
-			}
-
-			for _, arg := range transactionArguments {
-				err := tx.AddArgument(arg)
-
-				if err != nil {
-					cli.Exitf(1, "Failed to add %s argument to a transaction ", conf.Code)
-				}
-			}
-		}
-
 		signerRole := cli.SignerRole(conf.Role)
 		switch signerRole {
 		case cli.SignerRoleAuthorizer:
-			tx.AddAuthorizer(signerAccount.Address)
+			// Ignored if we're loading from a tx payload
+			allAuthorizers = []string{signerAccount.Address.String()}
 		case cli.SignerRolePayer:
 			if payer != signerAccount.Address {
 				cli.Exitf(1, "Role specified as Payer, but Payer address also provided, and different: %s !=", payer, signerAccount.Address)
 			}
 		case cli.SignerRoleProposer:
-			cli.Exitf(1, "Proposer role not yet supported: %s", conf.Role)
+			// Just sign payload, no special actions needed
 		default:
 			cli.Exitf(1, "unknown role %s", conf.Role)
 		}
 
-		for _, authorizerString := range conf.AdditionalAuthorizers {
-			authorizerAddress := flow.HexToAddress(authorizerString)
-			tx.AddAuthorizer(authorizerAddress)
+		if conf.Payload != "" && conf.Code != "" {
+			cli.Exitf(1, "Both a partial transaction and Cadence code file provided, but cannot use both")
+		} else if conf.Payload != "" {
+			tx = utils.LoadTransactionPayloadFromFile(conf.Payload)
```

I maybe wouldn't silently ignore args and payer here.